### PR TITLE
[fix](TabletInvertedIndex) fix potential deadlock between ForkJoinPool and TabletInvertedIndex

### LINKED ISSUE #11319: [Bug](TabletInvertIndex) Fe dead Lock with unknown reseaon
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

1.1/master

### What's Wrong?

Master hang because of Dead lock

### What You Expected?

Master Fe not hang

### How to Reproduce?

many thread waiting for 0x000010011ca0d7a0.
```
  1729 "tablet stat mgr" #38 daemon prio=5 os_prio=0 cpu=250439.25ms elapsed=55611.53s tid=0x00007f1ea800ba40 nid=0x45410 waiting on condition  [0x00007f1f918c9000]
  1730    java.lang.Thread.State: WAITING (parking)
  1731         at jdk.internal.misc.Unsafe.park(java.base@12.0.1/Native Method)
  1732         - parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
  1733         at java.util.concurrent.locks.LockSupport.park(java.base@12.0.1/LockSupport.java:194)
  1734         at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(java.base@12.0.1/AbstractQueuedSynchronizer.java:885)
  1735         at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireShared(java.base@12.0.1/AbstractQueuedSynchronizer.java:1009)
  1736         at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireShared(java.base@12.0.1/AbstractQueuedSynchronizer.java:1324)
  1737         at java.util.concurrent.locks.ReentrantReadWriteLock$ReadLock.lock(java.base@12.0.1/ReentrantReadWriteLock.java:738)
  1738         at org.apache.doris.catalog.TabletInvertedIndex.readLock(TabletInvertedIndex.java:101)
  1739         at org.apache.doris.catalog.TabletInvertedIndex.getReplica(TabletInvertedIndex.java:448)
  1740         at org.apache.doris.catalog.TabletStatMgr.updateTabletStat(TabletStatMgr.java:129)
  1741         at org.apache.doris.catalog.TabletStatMgr.lambda$runAfterCatalogReady$0(TabletStatMgr.java:63)
  1742         at org.apache.doris.catalog.TabletStatMgr$$Lambda$797/0x00007f1e58b4c0b0.accept(Unknown Source)
  1743         at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(java.base@12.0.1/ForEachOps.java:183)
  1744         at com.google.common.collect.CollectSpliterators$1WithCharacteristics.lambda$forEachRemaining$1(CollectSpliterators.java:67)
  1745         at com.google.common.collect.CollectSpliterators$1WithCharacteristics$$Lambda$799/0x00007f1e58b4b4b0.accept(Unknown Source)
  1746         at java.util.stream.Streams$RangeIntSpliterator.forEachRemaining(java.base@12.0.1/Streams.java:104)
  1747         at com.google.common.collect.CollectSpliterators$1WithCharacteristics.forEachRemaining(CollectSpliterators.java:67)
  1748         at java.util.stream.AbstractPipeline.copyInto(java.base@12.0.1/AbstractPipeline.java:484)
  1749         at java.util.stream.ForEachOps$ForEachTask.compute(java.base@12.0.1/ForEachOps.java:290)
  1750         at java.util.concurrent.CountedCompleter.exec(java.base@12.0.1/CountedCompleter.java:746)
  1751         at java.util.concurrent.ForkJoinTask.doExec(java.base@12.0.1/ForkJoinTask.java:290)
  1752         at java.util.concurrent.ForkJoinTask.doInvoke(java.base@12.0.1/ForkJoinTask.java:408)
  1753         at java.util.concurrent.ForkJoinTask.invoke(java.base@12.0.1/ForkJoinTask.java:736)
  1754         at java.util.stream.ForEachOps$ForEachOp.evaluateParallel(java.base@12.0.1/ForEachOps.java:159)
  1755         at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(java.base@12.0.1/ForEachOps.java:173)
  1756         at java.util.stream.AbstractPipeline.evaluate(java.base@12.0.1/AbstractPipeline.java:233)
  1757         at java.util.stream.ReferencePipeline.forEach(java.base@12.0.1/ReferencePipeline.java:497)
  1758         at java.util.stream.ReferencePipeline$Head.forEach(java.base@12.0.1/ReferencePipeline.java:661)
  1759         at org.apache.doris.catalog.TabletStatMgr.runAfterCatalogReady(TabletStatMgr.java:54)
  1760         at org.apache.doris.common.util.MasterDaemon.runOneCycle(MasterDaemon.java:58)
  1761         at org.apache.doris.common.util.Daemon.run(Daemon.java:116)
  1762 
  1763    Locked ownable synchronizers:
  1764         - None
```

But no thread holds 0x000010011ca0d7a0;
```
 cat 283354_6  | grep 0x000010011ca0d7a0
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	- parking to wait for  <0x000010011ca0d7a0> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)

```

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by caiconghui**
# Proposed changes

Issue Number: close #11319 

## Problem Summary:
The default  ForkJoinPool is shared by all parallelStream by default, and we obtain  read lock outside the ForkJoinPool in TabletInvertIndex while we obtain read lock inside the same ForkJoinPool in TabletStatMgr which may cause deadlock

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/catalog/TabletInvertedIndex.java [hunk 1]
```diff
@@ -128,128 +131,131 @@ public void tabletReport(long backendId, Map<Long, TTablet> backendTablets,
...
-                        tabletFoundInMeta.add(tabletId);
-                        TTabletInfo backendTabletInfo = backendTablet.getTabletInfos().get(0);
-                        if (partitionIdInMemorySet.contains(
-                                backendTabletInfo.getPartitionId()) != backendTabletInfo.isIsInMemory()) {
-                            synchronized (tabletToInMemory) {
-                                tabletToInMemory.add(new ImmutableTriple<>(tabletId, backendTabletInfo.getSchemaHash(),
-                                        !backendTabletInfo.isIsInMemory()));
+                taskPool.submit(() -> {
+                    // traverse replicas in meta with this backend
+                    replicaMetaWithBackend.entrySet().parallelStream().forEach(entry -> {
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/TabletInvertedIndex.java:136]: I have a question, under what amount of data does parallelStream+synchronized performance more than Stream

Comment (caiconghui) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/TabletInvertedIndex.java:136]: it depends on both task number and the time cost per task, need to do some experiment and think about whether improve performance significantly

**Comment by lide-reed** [comment]
LGTM

**Review by lide-reed — COMMENTED** [review]
LGTM

**Review by lide-reed — APPROVED** [review]
LGTM

**Pull request merged**